charts/agent-stack-k8s/templates/deployment.yaml.tpl

```
@@ -10,7 +10,7 @@ spec:
   template:
     metadata:
       labels:
-        app: {{ .Release.Name }}
+        {{- template "mychart.labels" }}
```

I think `include` is what's needed here
```suggestion
        {{- include "mychart.labels" }}
```